valid-jsdoc incorrectly flags multi-level params as errors

Given JSDoc such as this:

``` js
/**
 * Traverses multiple directories and calls a callback on each file.
 * @param {Object} options The option for the traversal.
 * @param {string[]} options.files An array of file and directory paths to traverse.
 * @param {string[]} options.exclude An array of file and directory paths to ignore.
 * @param {Function} callback A function to call for each file.
 * @returns {void}
 */
```

The output is:

```
  83:0  error  Expected JSDoc for 'callback' but found 'options.files'  valid-jsdoc
```

We should skip over any named parameters with a dot, as these are properties of an object that is a parameter. There should be no warning in this case.

--- a/lib/rules/valid-jsdoc.js
+++ b/lib/rules/valid-jsdoc.js
@@ -102,7 +102,7 @@ module.exports = function(context) {
 
                         if (params[tag.name]) {
                             context.report(jsdocNode, "Duplicate JSDoc parameter '{{name}}'.", { name: tag.name });
-                        } else {
+                        } else if (tag.name.indexOf(".") === - 1){
                             params[tag.name] = 1;
                         }
                         break;

--- a/tests/lib/rules/valid-jsdoc.js
+++ b/tests/lib/rules/valid-jsdoc.js
@@ -28,6 +28,7 @@ eslintTester.addRuleTest("lib/rules/valid-jsdoc", {
         "/**\n* Description\n* @param {Object} p bar\n* @param {string} p.name bar\n* @returns {string} desc */\nFoo.bar = function(p){};",
         "(function(){\n/**\n* Description\n* @param {string} p bar\n* @returns {string} desc */\nfunction foo(p){}\n}())",
         "var o = {\n/**\n* Description\n* @param {string} p bar\n* @returns {string} desc */\nfoo: function(p){}\n};",
+        "/**\n* Description\n* @param {Object} p bar\n* @param {string[]} p.files qux\n* @param {Function} cb baz\n* @returns {void} */\nfunction foo(p, cb){}",
         {
             code: "/**\n* Description\n* @return {void} */\nfunction foo(){}",
             args: [1, {}]

